cmd/cluster/core/dump.go

```diff
@@ -190,7 +193,55 @@ func dumpGuestCluster(ctx context.Context, opts *DumpOptions) error {
 		return fmt.Errorf("failed to get hosted cluster %s/%s: %w", opts.Namespace, opts.Name, err)
 	}
 	cpNamespace := manifests.HostedControlPlaneNamespace(opts.Namespace, opts.Name)
-	localPort := rand.Intn(45000-32767) + 32767
+
+	var localPort int
+	var forwarderStop chan struct{}
+
+	if opts.DumpGuestClusterThroughKubeService {
+		localPort = -1
```

Comment:
Can you add a comment here explaining why this is -1? Right now it might not be clear to a user.

Maybe something like `// Indicates connection via kube-apiserver service instead of port-forward`